tree_arena/src/tree_arena_safe.rs

```diff
@@ -1,15 +1,10 @@
 // Copyright 2024 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
-//! This module will eventually be factored out into a separate crate.
-//!
-//! In the meantime, we intentionally don't make the types in this module part of
-//! our public API, but still implement methods that a standalone crate would have.
-//!
 //! The types defined in this module don't *actually* implement an arena. They use
-//! 100% safe code, which has a significant performance overhead. The final version
-//! will use an arena and unsafe code, but should have the exact same exported API as
-//! this module.
+//! 100% safe code, which has a significant performance overhead.
+//! The version in `tree_arena_unsafe.rs` uses unsafe code, but should have the
+//! exact same exported API as this module.
```

Comment:
I wonder if we can `#![forbid(unsafe_code)]` in this module. Needn't be in this PR

Reply:
Probably, yeah.